### DISCUSSION TITLE
cant launch webui last update

### DISCUSSION THREAD

**Discussion by NaotoEnjoyer** [TheLastBen/fast-stable-diffusion#1138] [Q&A] [answered]
here's the error,so i have to enable all sections even install repo which ofc git pulls each time i launch it thats because it just doesnt launch if i dont enable everything in order, how i got stuck with this error
Traceback (most recent call last):
File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 11, in
from modules import import_hook
ImportError: cannot import name 'import_hook' from 'modules' (unknown location)

after i deleted sd repo folder from my gdrive completely to do fresh install in colab i get this:
/usr/local/lib/python3.8/dist-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.16.5 and <1.23.0 is required for this version of SciPy (detected version 1.24.0
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"
Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 195, in <module>
    webui()
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 133, in webui
    initialize()
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 63, in initialize
    modules.sd_models.load_model()
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_models.py", line 284, in load_model
    checkpoint_info = checkpoint_info or select_checkpoint()
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_models.py", line 115, in select_checkpoint
    if len(model_checkpoint) == 0:
TypeError: object of type 'NoneType' has no len()
P.S i tested webui on my pc and update works fine

**Comment by TheLastBen** [comment, marked as answer]
remove the folder sd from your gdrive

**Reply by NaotoEnjoyer** [reply]
i removed sd folder completely and on new colab launch procedure i still get last error i posted it creates sd repo on install webui repo section and on last section start webui gives this no matter what i pick use gradio, not

**Reply by NaotoEnjoyer** [reply]
my dumbass shouldhave just wait, someone just posted on webui git page that this import hook error was easily fixed by just adding this missing file in modules, and now that ive deleted sd folder completely.. ugh

**Reply by TheLastBen** [reply]
you can find it in the trash if you didn't remove it

**Reply by NaotoEnjoyer** [reply]
so i tried again after i made sure i removed it completely, still got it, and just now i tried to launch colab on completely other clean google drive account, literally nothing is there and i still got this error by the end of colab
/usr/local/lib/python3.8/dist-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.16.5 and <1.23.0 is required for this version of SciPy (detected version 1.23.0
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"
Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 195, in <module>
    webui()
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 133, in webui
    initialize()
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 63, in initialize
    modules.sd_models.load_model()
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_models.py", line 284, in load_model
    checkpoint_info = checkpoint_info or select_checkpoint()
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/sd_models.py", line 115, in select_checkpoint
    if len(model_checkpoint) == 0:
TypeError: object of type 'NoneType' has no len()

**Reply by NaotoEnjoyer** [reply]
something is wrong not with gdrive it seems

**Reply by mimondica** [reply]
> my dumbass shouldhave just wait, someone just posted on webui git page that this import hook error was easily fixed by just adding this missing file in modules, and now that ive deleted sd folder completely.. ugh

I had the exact same issue, fresh installed and got the same error but I had a safe copy of my old sd folder. I imported the module as you described and it works, however the numpy error message still appears

**Reply by NaotoEnjoyer** [reply]
> you can find it in the trash if you didn't remove it

so, i created new fresh google account, had empty gdrive, did go thru regular colab launch procedure, still got this issue
usr/local/lib/python3.8/dist-packages/scipy/__init__.py:146: UserWarning: A NumPy version >=1.16.5 and <1.23.0 is required for this version of SciPy (detected version 1.23.0
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion}"
  something is clearly wrong with either colab or today's webui update. even completely fresh setup fails

**Reply by TheLastBen** [reply]
https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/5974

**Reply by i-huzaifa-arshad** [reply]
I also am having the numpy error

**Reply by NaotoEnjoyer** [reply]
man its super weird, thats it, im 100% sure its colab or webui issue rn, yesterday everything worked fine, now i cant even install colab and all stuff it needs on new google drive account that doesnt has anything on it, hellaweirdchamp.. im so confused getting this error constantly

**Reply by NaotoEnjoyer** [reply]
ok ive read linked threads, but how do i even git reverse if i even cant install sd and launch it properly now?

**Reply by NaotoEnjoyer** [reply]
> [AUTOMATIC1111/stable-diffusion-webui#5974](https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/5974)

ben you rly need to think on how to fix this, its rly weird, when it doesnt work like i and some other ppl explained, i updated my local webui rn and it works completely fine with this new update when your colab cant even launch it on new account anymore

**Reply by TheLastBen** [reply]
https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/5971

A1111 will fix it soon

**Reply by NaotoEnjoyer** [reply]
> [AUTOMATIC1111/stable-diffusion-webui#5971](https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/5971)
> 
> A1111 will fix it soon

ok thx sry, im a little frustrated after trying to launch it with 100500 diff attempts if i seemed rude sry again

**Reply by TheLastBen** [reply]
no worries
